Packs/GenericAPIEventCollector/Integrations/GenericAPIEventCollector/GenericAPIEventCollector_test.py

```diff
@@ -0,0 +1,417 @@
+from datetime import datetime
+from unittest.mock import patch
+
+import pytest
+
+from CommonServerPython import DemistoException
+from GenericAPIEventCollector import (
+    datetime_to_timestamp_format, timestamp_format_to_datetime, recursive_replace,
+    get_time_field_from_event_to_dt, is_pagination_needed, iso8601_to_datetime_str, try_load_json,
+    parse_json_param, generate_authentication_headers,
+    extract_pagination_params, PaginationLogic, TimestampFieldConfig, organize_events_to_xsiam_format, setup_search_events,
+    RequestData, generate_headers
+)
+
+
+def test_datetime_to_timestamp_format():
+    dt = datetime(2023, 5, 1, 12, 0, 0)
+    assert datetime_to_timestamp_format(dt, '%Y-%m-%dT%H:%M:%SZ') == '2023-05-01T12:00:00Z'
+    assert datetime_to_timestamp_format(dt, 'epoch') == str(dt.timestamp())
+
+
+def test_timestamp_format_to_datetime():
+    dt_str = '2023-05-01T12:00:00Z'
+    assert timestamp_format_to_datetime(dt_str, '%Y-%m-%dT%H:%M:%SZ') == datetime(2023, 5, 1, 12, 0, 0)
+    epoch_str = str(datetime(2023, 5, 1, 12, 0, 0).timestamp())
+    assert timestamp_format_to_datetime(epoch_str, 'epoch') == datetime(2023, 5, 1, 12, 0, 0)
+
+
+def test_recursive_replace():
+    org_dict = {'key1': 'value1', 'key2': {'key3': 'value3'}}
```

Comment:
Don't we wat to also check the case where 
    org_dict = {'key1': 'value', 'key2': {'key3': 'value'}}
    substitutions = [('value', 'new_value1')]